### PR TITLE
add tests for copy and move operations of the Handle class

### PR DESCRIPTION
**Pull request by saikishor**

### CODE REVIEW

File: hardware_interface/test/test_handle.cpp [hunk 1]
```diff
@@ -95,3 +95,51 @@ TEST(TestHandle, interface_description_command_interface_name_getters_work)
...
+  double value = 1.337;
+  double value_2 = 2.337;
+  hardware_interface::Handle handle{JOINT_NAME, FOO_INTERFACE, &value};
+  hardware_interface::Handle copy{JOINT_NAME, "random", &value_2};
+  EXPECT_DOUBLE_EQ(copy.get_value(), value_2);
+  EXPECT_DOUBLE_EQ(handle.get_value(), value);
+  copy = handle;
+  EXPECT_DOUBLE_EQ(copy.get_value(), value);
+  EXPECT_NO_THROW(copy.set_value(0.0));
+  EXPECT_DOUBLE_EQ(copy.get_value(), 0.0);
```
[the remark this answers is not in the document]
Comment (saikishor) [reply hardware_interface/test/test_handle.cpp:130]: Good one. Let's try.

[the remark this answers is not in the document]
Comment (saikishor) [reply hardware_interface/test/test_handle.cpp:133]: I just pushed the change. Locally it worked for me

[the remark this answers is not in the document]
Comment (Wiktor-99) [reply hardware_interface/test/test_handle.cpp:133]: But setting the value in the copy should modify the original handle? I think this was the problem

[the remark this answers is not in the document]
Comment (saikishor) [reply hardware_interface/test/test_handle.cpp:133]: You are right. I missed the logic there. It shouldn't be that way.

[the remark this answers is not in the document]
Comment (Wiktor-99) [reply hardware_interface/test/test_handle.cpp:133]: You could try the solution that I proposed in the issue, I think it might help.

[the remark this answers is not in the document]
Comment (saikishor) [reply hardware_interface/test/test_handle.cpp:133]: Yes, there is one corner case as well. I'll fix it once I reach home after work. I'll keep you posted. Thanks

[the remark this answers is not in the document]
Comment (saikishor) [reply hardware_interface/test/test_handle.cpp]: @Wiktor-99 done. I've added the new fix and also updated the tests

File: hardware_interface/include/hardware_interface/handle.hpp [hunk 2]
```diff
@@ -45,6 +45,7 @@ class Handle
     handle_name_(prefix_name_ + "/" + interface_name_),
     value_ptr_(value_ptr)
   {
+    value_ = std::monostate{};
```
Comment (Wiktor-99) [inline hardware_interface/include/hardware_interface/handle.hpp:48]: We have two other constructors which do not set this member, we could do default member initialization. 
```
HANDLE_DATATYPE value_ = std::monostate;
```

File: hardware_interface/include/hardware_interface/handle.hpp [hunk 3]
```diff
@@ -104,7 +112,14 @@ class Handle
...
       value_ = other.value_;
-      value_ptr_ = other.value_ptr_;
+      if (std::holds_alternative<std::monostate>(value_))
+      {
+        value_ptr_ = other.value_ptr_;
+      }
+      else
+      {
+        value_ptr_ = std::get_if<double>(&value_);
+      }
```
Comment (Wiktor-99) [inline hardware_interface/include/hardware_interface/handle.hpp:122]: We have code duplication, maybe we could do method?

File: hardware_interface/include/hardware_interface/handle.hpp (hunk 2 above)
Comment (saikishor) [reply hardware_interface/include/hardware_interface/handle.hpp:48]: Sure. It makes more sense this way. I will do that

File: hardware_interface/include/hardware_interface/handle.hpp (hunk 3 above)
Comment (saikishor) [reply hardware_interface/include/hardware_interface/handle.hpp:122]: Here, this is a copy constructor/assignment operation, most of the code is kinda duplicated. If it's okay, let's leave it this way and then in future, I want to see it we can remove some.

I had to implemented all 5 due to the rule of five.

What do you think?

Comment (Wiktor-99) [reply hardware_interface/include/hardware_interface/handle.hpp:122]: I'm fine with that, maybe copy-swap idiom might help

Comment (saikishor) [reply hardware_interface/include/hardware_interface/handle.hpp]: Done

File: hardware_interface/include/hardware_interface/handle.hpp (hunk 2 above)
Comment (saikishor) [reply hardware_interface/include/hardware_interface/handle.hpp]: Done

**Review by Wiktor-99 — APPROVED** [review]
LGTM

**Comment by saikishor** [comment]
@Wiktor-99 thanks for the nice suggestions! :)

**Review by christophfroehlich — APPROVED** [review]
I always learn something new with you guys :D LGTM as far as I can judge

**Review by bmagyar — APPROVED** [review]
very professional and definitely much needed on the most commonly used object type in the framework!
